skia/RSkComponentRNSVGSvgView.cpp

```diff
@@ -0,0 +1,192 @@
+
+    if (!viewBox_.isEmpty()) {
+
+      // A viewBox overrides the intrinsic viewport.
+      viewPort = SkSize::Make(viewBox_.width(), viewBox_.height());
+
+      contentMatrix.preConcat(
+        SkMatrix::MakeRectToRect(viewBox_, viewPortRect, SkMatrix::kFill_ScaleToFit));
+    }
+
+    if (!contentMatrix.isIdentity()) {
+        ctx->saveOnce();
+        ctx->canvas()->concat(contentMatrix);
+    }
+
+    if (viewPort != ctx->lengthContext().viewPort()) {
+        ctx->writableLengthContext()->setViewPort(viewPort);
+    }
+    return this->INHERITED::onPrepareToRender(ctx);
+}
+
+
+void RSkComponentRNSVGSvgView::onSetAttribute(SkSVGAttribute attr, const SkSVGValue& v) {
+    switch (attr) {
+    case SkSVGAttribute::kX:
+        if (const auto* x = v.as<SkSVGLengthValue>()) {
+            x_ = *x;
+        }
+        break;
+    case SkSVGAttribute::kY:
+        if (const auto* y = v.as<SkSVGLengthValue>()) {
+            y_ = *y;
+        }
+        break;
+    case SkSVGAttribute::kWidth:
+        if (const auto* w = v.as<SkSVGLengthValue>()) {
+            width_ = *w;
+        }
+        break;
+    case SkSVGAttribute::kHeight:
+        if (const auto* h = v.as<SkSVGLengthValue>()) {
+            height_ = *h;
+        }
+        break;
+
+    case SkSVGAttribute::kViewBox:
+        if (const auto* vb = v.as<SkSVGViewBoxValue>()) {
+            viewBox_=*vb;
+        }
+        break;
+    default:
+        this->INHERITED::onSetAttribute(attr, v);
+    }
+}
+
+SkSize RSkComponentRNSVGSvgView::getContainerSize(const SkSVGLengthContext& lctx) const {
+    // Percentage values do not provide an intrinsic size.
+    if (width_.unit() == SkSVGLength::Unit::kPercentage ||
+        height_.unit() == SkSVGLength::Unit::kPercentage) {
+        return SkSize::Make(0, 0);
+    }
+
+    return SkSize::Make(lctx.resolve(width_, SkSVGLengthContext::LengthType::kHorizontal),
+                        lctx.resolve(height_, SkSVGLengthContext::LengthType::kVertical));
+}
+
+inline void RSkComponentRNSVGSvgView::alterSkiaDefaultPaint() {
+ //Note: Skia's default paint is black. WhereIn on SVG REference platform it's transparent
+ //      To Match with Reference output, for the next immendiate child of SVGView,
+ //       resetting color to transparent, if color not speciifed
+
+  for (auto item:childRSkNodeList_) {
+    RNS_LOG_INFO(" Child Tag : "<<(int)item->tag());
+    if(item->tag() == SkSVGTag::kG) {
+      auto component =static_cast<RSkComponentRNSVGGroup *>(item.get())->getComponentData();
+      auto const &newRNSVGGroupPropsProps = *std::static_pointer_cast<RNSVGGroupProps const>(component.props);
+      auto rskSVGNode =static_cast<RSkSVGNode *>(item.get());
+      // Reset Stroke & Fill Color if not specifed to transparent
+      if(std::find (newRNSVGGroupPropsProps.propList.begin(), newRNSVGGroupPropsProps.propList.end(), "fill") == newRNSVGGroupPropsProps.propList.end()) {
```

Comment:
can the both iteration of find on the propList be clubbed

Reply:
As it is just two, to avoid iterating over complete list & for better clarity on implementation handled it this way.